test/Benchmarks/src/Decimal_Benchmark.enso

```diff
@@ -0,0 +1,67 @@
+from Standard.Base import all
+
+from Standard.Test import Bench, Faker
+
+
+options = Bench.options . set_warmup (Bench.phase_conf 1 3) . set_measure (Bench.phase_conf 1 3)
+
+type Lazy
+    Value ~value
+
+type Data
+    Value float_pairs decimal_pairs int_pairs decimal_int_pairs small_float_int_pairs small_decimal_int_pairs
+
+    new faker =
+        num_pairs = 10000
+
+        floats0 = Vector.new num_pairs _-> faker.float_full_range_exp
+        floats1 = Vector.new num_pairs _-> faker.float_full_range_exp
+        ints0 = Vector.new num_pairs _-> faker.integer -500000000 500000000
+        ints1 = Vector.new num_pairs _-> faker.integer -500000000 500000000
+        small_floats = Vector.new num_pairs _-> faker.float -1000 1000
+        small_ints = Vector.new num_pairs _-> faker.integer 0 90
+
+        decimals0 = floats0.map (f-> Decimal.from_float f)
+        decimals1 = floats1.map (f-> Decimal.from_float f)
+        decimal_ints0 = ints0.map (i-> Decimal.from_integer i)
+        decimal_ints1 = ints1.map (i-> Decimal.from_integer i)
+        small_decimals = small_floats.map (f-> Decimal.from_float f)
+
+        Data.Value (floats0.zip floats1) (decimals0.zip decimals1) (ints0.zip ints1) (decimal_ints0.zip decimal_ints1) (small_floats.zip small_ints) (small_decimals.zip small_ints)
+
+collect_benches = Bench.build builder->
+    ## No specific significance to this constant, just fixed to make generated set deterministic
+    fixed_random_seed = 1644575867
+    faker = Faker.new fixed_random_seed
+
+    lazy_data = Lazy.Value (Data.new faker)
+
+    builder.group "Float_Decimal_comparison" options group_builder->
+        data = lazy_data.value
+        float_pairs = data.float_pairs
+        float_pairs_without_zero_second = float_pairs.filter (pr-> (pr.at 1) != 0.0)
+        decimal_pairs = data.decimal_pairs
+        decimal_pairs_without_zero_second = decimal_pairs.filter (pr-> (pr.at 1) != 0.0)
+        int_pairs = data.int_pairs
+        decimal_int_pairs = data.decimal_int_pairs
+        small_float_int_pairs = data.small_float_int_pairs
+        small_decimal_int_pairs = data.small_decimal_int_pairs
```

Comment:
This is not the correct usage of the lazy pattern.

Note that each of these operations actually _forces_ the lazy field and evaluates it _inside of the group builder_.

But the group builder is run when gathering _what_ tests will be run, before running them. So this may trigger the costly initialization even when no benchmarks are run.

---

Given these are benchmarks, I'm not sure how important that is to ensure that just gathering benchmark structure is fast. Ideally it should be done like with our tests, but I feel like it's less important.

Still, we should either use the lazy pattern correctly OR not use it at all. Currently, we are pretending like we do stuff lazily, but we actually eagerly force it immediately, so there's no real benefit to this laziness.

To fix it, I guess we can wrap all expressions in this marked block with `Lazy.Value`, then it will get only executed when actually needed.

I'm wondering if we should add sth like this to our stdlib, as it's useful in places like this.

Reply:
Yes, I think this would be a useful utility, something like:

```
type Lazy
    Value ~eval

lazy ~thunk = Lazy.Value thunk

a = lazy (big_computation 1 2 3)
b = a.eval
```

(Perhaps `eval` should be just `value` or `compute`.)

However, this still has the inconvenience of having to remember to call `.eval`. At some point it would be good if the engine could have an internal lazy value type that would automatically be evaluated when necessary.

Reply:
Rewrote to make it actually lazy (and I confirmed it with `printlins`).

I think I was unclear exactly when lazy values are computed, but it is clearer to me now.